acceptance/builder_test.go

```diff
@@ -0,0 +1,418 @@
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/invalid_buildpack_api_group.toml",
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to : parse buildpack API"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+		})
+
+		// .../cmd/lifecycle/builder.go#readData
+		when("plan.toml", func() {
+			when("not found", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to parse detect plan"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+			//TODO: check some output file for this case not except any error message
+			when("empty", func() {
+				it("success", func() {
+					h.DockerRunAndCopy(t,
+						containerName,
+						copyDir,
+						ctrPath("/layers"),
+						builderImage,
+						h.WithFlags(
+							"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+							"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/empty_plan.toml",
+							"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						),
+					)
+					// check builder metadata.toml for success test
+					assertBuilderMetadata(t, filepath.Join(copyDir, "layers/config/metadata.toml"))
+				})
+			})
+
+			when("invalid", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/wrong_plan.toml",
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to parse detect plan: Near line"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+		})
+	})
+
+	when("determining the location of input files", func() {
+		// .../cmd/lifecycle/builder.go#Args
+		when("group.toml path is not specified", func() {
+			it("will look for group.toml in the provided layers directory", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_LAYERS_DIR=/layers/different_layer_dir_from_env",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					builderImage,
+				)
+				_, err := command.CombinedOutput()
+				h.AssertNil(t, err)
+				//expected := "failed to read buildpack group: open /layers/group.toml"
```

Comment:
Ah - so if the layers dir is set to `/layers/different_layer_dir_from_env` the builder is going to try to use `/layers/different_layer_dir_from_env/group.toml`. If you put a real group file in your fixture (acceptance/testdata/builder/container/layers/different_layer_dir_from_env/group.toml) you could see that the file was used by either (a) referencing a non-existent buildpack, and checking the error message, or (b) referencing a real buildpack, but a different buildpack from `layers/group.toml`, and checking the output from the buildpack's bin/build. I would lean toward (b), but I think either could work.

Reply:
we need to add `metadata.toml` checking.

Reply:
different buildpack is added. then metadata file is checked.